src/lib.rs

```diff
@@ -48,6 +48,43 @@ pub enum Error {
     Signal(&'static str),
 }
 
+enum RawModeGuardState {
+    None,
+    #[cfg(target_family = "windows")]
+    PreInit {
+        stdin: std::os::windows::raw::HANDLE,
+        stdout: std::os::windows::raw::HANDLE,
+    },
+    #[cfg(target_family = "unix")]
+    PreInit(std::os::fd::RawFd),
+    Some(RawModeGuard),
```

Comment:
Putting `Some` and `None` in an enum that's not `Option` might be confusing, but it's not that bad

Reply:
Hm, so is the reason to not make it `Option<RawModeGuard>` with just two states because then you can't call `ensure_enabled` on it when it's `None`?

Reply:
that's fair 😅